File: 2500-1048-942-di-string-match/2500-1048-942-di-string-match.py
class Solution(object):
    def diStringMatch(self, s):
        result=[]
        i_counter=0
        j_counter=len(s)
        lengthOfS=len(s)
        for i in range(lengthOfS):
            if s[i]=='I':
                result.append(i_counter)
                i_counter+=1
            else:
                result.append(j_counter)
                j_counter-=1
        if s[len(s)-1]=='I':
            result.append(i_counter)
        else:
            result.append(j_counter)
        return result
        
        """
        :type s: str
        :rtype: List[int]
        """
        

File: 2500-1048-942-di-string-match/2500-1048-942-di-string-match.java
class Solution {
    public int[] diStringMatch(String s) {
        int[] result=new int[s.length()+1];
        int i_counter=0,j_counter=s.length();
        int lenghtOfS=s.length();
        for(int i=0;i<lenghtOfS;i++)
        {  if(s.charAt(i)=='I')
             { result[i]=i_counter;
                i_counter++;
             }
            else
            { result[i]=j_counter;
                j_counter--; 
            }
                
        }
        if(s.charAt(lenghtOfS-1)=='I')
         result[s.length()]=i_counter;
         result[s.length()]=j_counter;
        return result; 
    }
}